File: 常設中のコンテスト/アルゴリズムと数学 演習問題集/017 - Least Common Multiple of N Integers/lcm_n.go
package main

import (
    "bufio"
    "fmt"
    "os"
)

func gcd(a, b int) int  {
    if b == 0 {
        return a
    }
    return gcd(b, a % b)
}

func lcm(a, b int) int {
    return (b / gcd(a, b)) * a
}

func main()  {
    in := bufio.NewReader(os.Stdin)

    var n int
    fmt.Fscan(in, &n)

    nums := make([]int, n)
    for i := 0; i < n; i++ {
        fmt.Fscan(in, &nums[i])
    }

    ans := lcm(nums[0], nums[1])
    for i := 2; i < n; i++ {
        ans = lcm(ans, nums[i])
    }

    fmt.Printf("%v\n", ans)
}

File: 常設中のコンテスト/アルゴリズムと数学 演習問題集/017 - Least Common Multiple of N Integers/lcm_n.py
#!/usr/bin/env python3


def gcd(a, b):
    if b == 0:
        return a

    return gcd(b, a % b)

def lcm(a, b):
    return a * b // gcd(a, b)


if __name__ == "__main__":
    n = int(input())

    nums = list(map(int, input().split()))

    ans = lcm(nums[0], nums[1])
    for i in range(2, len(nums)):
        ans = lcm(ans, nums[i])

    print(ans)
